src/datanode/src/instance.rs

```diff
@@ -169,21 +170,27 @@ impl Instance {
     }
 }
 
-pub(crate) async fn new_object_store(store_config: &ObjectStoreConfig) -> Result<ObjectStore> {
+pub(crate) async fn new_object_store(opts: &DatanodeOptions) -> Result<ObjectStore> {
     // TODO(dennis): supports other backend
-    let data_dir = util::normalize_dir(match store_config {
+    let data_dir = match &opts.storage {
         ObjectStoreConfig::File { data_dir } => data_dir,
-    });
+    };
 
-    fs::create_dir_all(path::Path::new(&data_dir))
-        .context(error::CreateDirSnafu { dir: &data_dir })?;
+    let mut path = PathBuf::from(data_dir);
```

Comment:
It's too tricky, the user may be surprised that the data directory is not expected as they set.

I recommend to supports command line args `--wal-dir` and `--data-dir` , then we can modify the distributed getting start doc and pass these options by command line.